scripts/build.js

```diff
@@ -1,37 +1,44 @@
 import fs from 'node:fs';
-import assert from 'node:assert/strict';
 import simplifyRanges from 'simplify-ranges';
-
-const CATEGORIES = {
-	A: 'ambiguous',
-	F: 'fullwidth',
-	H: 'halfwidth',
-	N: 'neutral',
-	Na: 'narrow',
-	W: 'wide',
-};
-const DEFAULT_CATEGORY = CATEGORIES.N;
+import {outdent} from 'outdent';
+import indentString from 'indent-string';
+
+const CATEGORY_NAMES = new Map([
+	['A', 'ambiguous'],
+	['F', 'fullwidth'],
+	['H', 'halfwidth'],
+	['N', 'neutral'],
+	['Na', 'narrow'],
+	['W', 'wide'],
+]);
+const DEFAULT_CATEGORY = CATEGORY_NAMES.get('N');
 
 const toHexadecimal = number => number === 0 ? '0' : `0x${number.toString(16).toUpperCase()}`;
+const indent = string => indentString(string, 1, {indent: '\t'});
 
 function parse(input) {
-	// Remove comments
-	input = input.trim().replaceAll(/^#.*$/gm, '').trim();
-
-	const lines = input.split('\n');
-	const categories = new Map();
+	const categories = new Map(Array.from(CATEGORY_NAMES, ([, category]) => [category, []]));
 
 	// Parse input and group by category
-	for (const line of lines) {
-		const [codePoint, rest] = line.split(';').map(x => x.trim());
-		const type = rest.split(' ')[0];
-		assert.ok(Object.hasOwn(CATEGORIES, type));
-		const category = CATEGORIES[type];
-		const [start, end = start] = codePoint.split('..').map(part => Number.parseInt(part, 16));
-		if (!categories.has(category)) {
-			categories.set(category, []);
+	for (let line of input.split('\n')) {
+		// Remove comments
+		line = line.replace(/#.*$/, '').trim();
```

Comment:
I think it's nicer to just remove them up front, since the comments are only at the top anyway.

Reply:
Comments are on every line

https://github.com/sindresorhus/get-east-asian-width/blob/a53c089c5db2c6eec2e7a107b883d565f1481ea4/scripts/EastAsianWidth.txt#L42


After the `#`

Reply:
👍